api/mapping/services_rules.py

```diff
@@ -841,3 +843,101 @@ def remove_mapping_rules(request, scan_report_id):
     rules = MappingRule.objects.all().filter(scan_report__id=scan_report_id)
 
     rules.delete()
+
+
+def analyse_concepts(scan_report_id):
+    mapping_rules = (
+        MappingRule.objects.all()
+        .filter(scan_report_id=scan_report_id)
+        .values_list("concept__concept", flat=True)
+        .distinct()
+    )
+    all_mapping_rules = MappingRule.objects.exclude(
+        scan_report_id=scan_report_id
+    ).values_list("concept__concept", flat=True)
+    data = []
+    descendant_list = []
+    ancestors_list = []
+
+    for rule in mapping_rules:
+
+        try:
+            descendants = ConceptAncestor.objects.filter(ancestor_concept_id=rule)
+            ancestors = ConceptAncestor.objects.filter(descendant_concept_id=rule)
+
+            for descendant in descendants:
+                desc = descendant.descendant_concept_id
+                if (desc in all_mapping_rules) & (desc != rule):
+                    concept = Concept.objects.get(concept_id=desc)
+                    rule_name = Concept.objects.get(concept_id=rule).concept_name
+                    source_ids = (
+                        MappingRule.objects.filter(Q(concept__concept=desc))
```

Comment:
I think you don't need a `Q` object here
```suggestion
                        MappingRule.objects.filter(concept__concept=desc)
```